### DISCUSSION TITLE
Does Supabase `createSupabaseServerClient` recreate on every run in SvelteKit?

### DISCUSSION THREAD

**Discussion by DOZBORNE** [supabase/supabase#17135] [Questions] [answered]
Hey all! 

I'm a React/Angular dev and new to the Supabase and Svelte land. 

Question regarding Supabase client creation for the server. The [Supabase docs](https://supabase.com/docs/guides/auth/auth-helpers/sveltekit#creating-a-supabase-client) suggests that we create our Supabase server client like so,

```
// src/hooks.server.js
import { PUBLIC_SUPABASE_URL, PUBLIC_SUPABASE_ANON_KEY } from '$env/static/public'
import { createSupabaseServerClient } from '@supabase/auth-helpers-sveltekit'

export const handle = async ({ event, resolve }) => {
  event.locals.supabase = createSupabaseServerClient({
    supabaseUrl: PUBLIC_SUPABASE_URL,
    supabaseKey: PUBLIC_SUPABASE_ANON_KEY,
    event,
  })

...
// code to forward event here etc...
}
```

Seems reasonable. However, since hooks in Svelte run on (before) every request, this means that we're creating a new client (and therefore connection) on _every_ single request, from every user that hits our app. **_Is this intended behavior?_** It just seems like creating one instance, and then sharing it would be the appropriate thing to do here. However, I'm at a bit of a loss.

Performance-wise, is this okay? This seems pretty bad?

Thanks for any help!

Devin

**Comment by GaryAustin1** [comment, marked as answer]
Also answered on Discord.
There is no session connection maintained between Supabase clients and the Supabase server.  The session is just stored in local storage or cookies on the client and then passed to the Server on every call in a header as a JWT.  The REST API itself is stateless.

So creating extra clients or not in your code is just using memory and a bit of time to set up the client object.  CreateClient does not do a server call on its own unless the access token (jwt) needs to be refreshed.

**Reply by DOZBORNE** [reply]
Thank you! Yes! I appreciate the answer on discord and the follow up here, was planning to take it down, but will leave it up for others :).
